Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -843,11 +863,18 @@ bool MainWindow::RequestStop()
 
     if (confirm != QMessageBox::Yes)
     {
+      m_render_widget->SetWaitingForMessageBox(false);
```

Comment:
There is some code above preventing this code to run twice before the message box has been closed. This should be 100% safe.

Reply:
If you mean because this is called both in the if and in the else condition, it's because in the else, it needs to be after `SetCursorLockedOnNextActivation()`

Reply:
It should probably be documented in source then; because my first thought was "lets move it up".

Reply:
Just saw it. My immediate question when I see this kind of comment though: "Why?"
Can you add that concisely, or would it take up many lines to explain?

Reply:
True, it should be explained, the code is definately hard to understand is seems like I'm doing random stuff.
`SetCursorLockedOnNextActivation(false)` needs to be before `SetWaitingForMessageBox(false)` because `SetWaitingForMessageBox(false)` decides whether to lock the mouse or not when finally set to false (after it had been true).
In this specific case, if you user clicks: "No" (I don't want to quick the emulation), then we immediately capture the mouse again. While they click "Yes" (I want to quit/stop the emulation), we DON'T capture the mouse again, despite the render widget staying alive for a while, waiting for the emulation to be shut down.

I will explain in the comment.

Reply:
Just added a comment